lib/miq_automation_engine/engine/miq_ae_event.rb

```diff
@@ -31,10 +33,13 @@ def self.raise_synthetic_event(event, inputs, message = nil)
       aevent      = build_evm_event(event, inputs)
     end
 
-    MiqAeEngine.deliver(obj_type, obj_id, aevent, instance, nil, nil, message)
+    user = automate_user
+    MiqAeEngine.deliver(obj_type, obj_id, aevent, instance, nil, user.id, message)
   end
 
   def self.raise_evm_event(event_name, target, inputs = {}, _message = nil)
+    user = automate_user
```

Comment:
@mkanoor @kbrock 
Both VM and User have a relationship to Tenant. I assume a Vm and its user should point to the same tenant.
If it is possible to find out the user of an event at this point, then we can also find out the tenant here via the user.  
Is it right?